src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,23 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.InvalidDataException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
+        if (storageDao.get(user.getLogin()) == null) {
```

Comment:
This check is insufficient. It only checks if the user with the same login does not already exist in the storage, but does not validate the user object itself. Ensure that all user's parameters are validated according to the checklist.